src/ORM/DataList.php

```diff
@@ -943,10 +943,19 @@ public function setQueriedColumns($queriedColumns)
      * Filter this list to only contain the given Primary IDs
      *
      * @param array $ids Array of integers
+     * @throws InvalidArgumentException
      * @return $this
      */
     public function byIDs($ids)
     {
+        foreach ($ids AS $id) {
+            if (!is_int($id)) {
```

Comment:
Sure, I would use is_numeric of the check and then parse all params to _int_. 
The only disadvantage to that is, that a string value like "0.1" is numeric and would parse to 0 using _intval_.
In addition to that, according to the docs (http://php.net/manual/en/function.intval.php) it behaves a bit differently for PHP7.1 with values like "1e5".